Validations for empitness of all fields on SignUp

Meghatronics opened the pull request:
Fixed the two referenced issue.

Pulled the forgot_password.dart and firebase_forgot_password.dart files from Geek's original work on them.

Meghatronics committed:
Validations for empitness of all fields,

Meghatronics committed:
Bringing back Geek's work on forgot password

moyosolaa commented:
can you let the error messages be more explanatory, and advice users to try again when necessary like the internal error message and the check email error message, check email for what?

Meghatronics committed:
Forgot password success shows dialog instead of snackbar

Meghatronics commented:
> can you let the error messages be more explanatory, and advice users to try again when necessary like the internal error message and the check email error message, check email for what?

Updated. Please check

Meghatronics committed:
edit

bamiogunfemi reviewed (approved):
LGTM

Meghatronics committed:
Reinserted Geek's code for closing app
Adjusted dismount of loader on signIN for smoother transition

Meghatronics commented:
Reinserted the WillPopScope on dashboard. Work originally done by Geek.

Also adjusted the dismount of loader from sign in to wait for `checkForUser` future, for smoother transition

Meghatronics committed:
Removed remember me.
Changed clickable text from gestureDetector to inkResponse
defined width property for terms white box.

geektutor commented:
LGTM

Meghatronics commented:
> LGTM

Approve good sir
